drivers/rail/rail_drv.c

```diff
@@ -0,0 +1,798 @@
+        .channelSpacing = 2000000U,
+        .physicalChannelOffset = 1,
+        .channelNumberStart = 1,
+        .channelNumberEnd = 10,
+        .maxPower = RAIL_TX_POWER_MAX,
+        .attr = &radio_channel_entry_915
+    }
+#endif
+};
+
+#endif
+
+
+#if (RAIL_RADIO_BAND == 868)
+
+static const uint32_t ieee802154_config_863[] = RAIL_IEEE802154_CONFIG_868MHZ;
+static const uint32_t ieee802154_config_863_min[] = {
+    0xFFFFFFFFUL,
+};
+
+static const RAIL_ChannelConfig_t _rail_radio_channel_config = {
+    .phyConfigBase          = ieee802154_config_863,
+    .phyConfigDeltaSubtract = ieee802154_config_863_min,
+    .configs                = &radio_channel_entry[0],
+    .length                 = 1
+};
+#elif (RAIL_RADIO_BAND == 915)
+
+static const uint32_t ieee802154_config_863[] = RAIL_IEEE802154_CONFIG_915MHZ;
+static const uint32_t ieee802154_config_915_min[] = {
+    0xFFFFFFFFUL,
+};
+
+static const RAIL_ChannelConfig_t _rail_radio_channel_config = {
+    .phyConfigBase          = ieee802154_config_915,
+    .phyConfigDeltaSubtract = ieee802154_config_915_min,
+    .configs                = &radio_channel_entry[0],
+    .length                 = 1
+};
+#endif
+
+
+
+
+static const RAIL_IEEE802154_Config_t _rail_ieee802154_config = {
+    .addresses = NULL,
+    .ackConfig = {
+        .enable = true,                     // Turn on auto ACK for IEEE 802.15.4
+        .ackTimeout = 1200,                 // why 1200? docu says 54 symbols * 16 us/symbol = 864 us
+        .rxTransitions = {
+            .success = RAIL_RF_STATE_RX,    // docu says RAIL_RF_STATE_TX?
+            .error = RAIL_RF_STATE_RX       // ignored
+        },
+        .txTransitions = {
+            .success = RAIL_RF_STATE_RX,
+            .error = RAIL_RF_STATE_RX 		// ignored
+        }
+    },
+    .timings = {
+        .idleToRx = 100,
+        // Make txToRx slightly lower than desired to make sure we get to
+        // RX in time
+        .txToRx = 192 - 10,
+        .idleToTx = 100,
+        .rxToTx = 192,
+        .rxSearchTimeout = 0,
+        .txToRxSearchTimeout = 0
+    },
+    .framesMask = RAIL_IEEE802154_ACCEPT_STANDARD_FRAMES,
+    .promiscuousMode = false,
+    .isPanCoordinator = false
+};
+
+
+static const RAIL_CsmaConfig_t _rail_csma_config = RAIL_CSMA_CONFIG_802_15_4_2003_2p4_GHz_OQPSK_CSMA;
+
+
+// TODO use define to restrict to supported freq
+RAIL_DECLARE_TX_POWER_VBAT_CURVES(piecewiseSegments, curvesSg, curves24Hp, curves24Lp);
+
+
+//////////////////// BUFFER MANAGEMENT //////////////////////////////////////
+
+
+// tx buffer
+static uint8_t _transmit_buffer[128]; // IEEE802154_FRAME_LEN_MAX + 1
+
+////////////////////// LOKAL VARIABLES /////////////////////////////////////
+
+
+// ref to rail_t/ netdev_t struct for this driver
+// TODO howto distiguish between multible netdevs ?
+//    necessary for multiprotocol support etc.
+// Possible solution: array an rhandle as index, key.
+static rail_t *_rail_dev = NULL;
```

Comment:
@miri64 do you have a suggestion for this case?

Reply:
As far as I can see this variable is only used in `_rail_radio_event_handler`. Is there maybe some context you can provide in `RAIL_Handle_t` or `RAIL_Events_t`?

Reply:
The RAIL_Handle_t and RAIL_Events_t are defined in the gecko sdk / "driver blob", so, I can not add information to them. 
I have to use the given prototype for an interrupt handler as well. So I have to map the RAIL_Handle_t to the rail_t / netdev structure somehow.

Reply:
Still open, but the new gecko sdk api provides a RAIL_Handle_t so I can use it to distinguish between instances (later).